docs/source/developer/repomap.rst

```diff
@@ -0,0 +1,447 @@
+- ``numba/targets/cpu.py`` - Context for code gen on CPU
+- ``numba/targets/base.py`` - Base class for all target contexts
+- ``numba/targets/codegen.py`` - Driver for code generation
+- ``numba/targets/boxing.py`` - Boxing and unboxing for most data types
+- ``numba/targets/intrinsics.py`` - Utilities for converting LLVM intrinsics to other math calls
+- ``numba/targets/callconv.py`` - Implements different calling conventions for Numba-compiled functions
+- ``numba/targets/iterators.py`` - Iterable data types and iterators
+- ``numba/targets/hashing.py`` - Hashing algorithms
+- ``numba/targets/ufunc_db.py`` - Big table mapping types to ufunc implementations
+- ``numba/targets/setobj.py`` - Python set type
+- ``numba/targets/options.py`` - Container for options that control lowering
+- ``numba/targets/printimpl.py`` - Print function
+- ``numba/targets/smartarray.py`` - Smart array (deprecated)
+- ``numba/targets/cmathimpl.py`` - Python complex math module
+- ``numba/targets/optional.py`` - Special type representing value or ``None``
+- ``numba/targets/tupleobj.py`` - Tuples (statically typed as immutable struct)
+- ``numba/targets/mathimpl.py`` - Python ``math`` module
+- ``numba/targets/heapq.py`` - Python ``heapq`` module
+- ``numba/targets/registry.py`` - Registry object for collecting implementations for a specific target
+- ``numba/targets/imputils.py`` - Helper functions for lowering
+- ``numba/targets/builtins.py`` - Python builtin functions and operators
+- ``numba/targets/externals.py`` - Registers external C functions needed to link generated code
+- ``numba/targets/quicksort.py`` - Quicksort implementation used with list and array objects
+- ``numba/targets/mergesort.py`` - Mergesort implementation used with array objects
+- ``numba/targets/randomimpl.py`` - Python and NumPy ``random`` modules
+- ``numba/targets/npyimpl.py`` - Implementations of most NumPy ufuncs
+- ``numba/targets/slicing.py`` - Slice objects, and index calculations used in slicing
+- ``numba/targets/numbers.py`` - Numeric values (int, float, etc)
+- ``numba/targets/listobj.py`` - Python lists
+- ``numba/targets/fastmathpass.py`` - Rewrite pass to add fastmath attributes to function call sites and binary operations
+- ``numba/targets/removerefctpass.py`` - Rewrite pass to remove unnecessary incref/decref pairs
+- ``numba/targets/cffiimpl.py`` - CFFI functions
+- ``numba/targets/descriptors.py`` - empty base class for all target descriptors (is this needed?)
+- ``numba/targets/arraymath.py`` - Math operations on arrays (both Python and NumPy)
+- ``numba/targets/linalg.py`` - NumPy linear algebra operations
+- ``numba/targets/rangeobj.py`` - Python `range` objects
+- ``numba/targets/npyfuncs.py`` - Kernels used in generating some NumPy ufuncs
+- ``numba/targets/arrayobj.py`` - Array operations (both NumPy and buffer protocol)
+- ``numba/targets/enumimpl.py`` - Enum objects
+- ``numba/targets/polynomial.py`` - ``numpy.roots`` function
```

Comment:
Should probably implement the rest of the polynomial function support. Note to self. Write tickets.